Formula/percona-server-mongodb.rb

```diff
@@ -0,0 +1,141 @@
+require "language/go"
+
+class PerconaServerMongodb < Formula
+  desc "Drop-in MongoDB replacement"
+  homepage "https://www.percona.com"
+  url "https://www.percona.com/downloads/percona-server-mongodb-3.4/percona-server-mongodb-3.4.2-1.2/source/tarball/percona-server-mongodb-3.4.2-1.2.tar.gz"
+  version "3.4.2-1.2"
+  sha256 "85d60cdbb8d7aed1625ac3840466489e8f54ea1d3311d5556a3b3736c4f0fd60"
+
+  option "with-boost", "Compile using installed boost, not the version shipped with mongodb"
+  option "with-sasl", "Compile with SASL support"
+
+  depends_on "boost" => :optional
+  depends_on "go" => :build
+  depends_on :macos => :mountain_lion
+  depends_on "scons" => :build
+  depends_on "openssl" => :recommended
+
+  conflicts_with "mongodb",
+    :because => "percona-server-mongodb and mongodb install the same binaries."
+
+  go_resource "github.com/mongodb/mongo-tools" do
+    url "https://github.com/mongodb/mongo-tools.git",
+        :tag => "r3.4.2",
+        :revision => "17fbdf31abca50cdfe27482b05b1476f42ecab0a",
+        :shallow => false
+  end
+
+  needs :cxx11
+
+  def install
+    ENV.cxx11 if MacOS.version < :mavericks
+    ENV.libcxx if build.devel?
+
+    # New Go tools have their own build script but the server scons "install" target is still
+    # responsible for installing them.
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    cd "src/github.com/mongodb/mongo-tools" do
+      args = %w[]
+
+      if build.with? "openssl"
+        args << "ssl"
+        ENV["LIBRARY_PATH"] = Formula["openssl"].opt_lib
+        ENV["CPATH"] = Formula["openssl"].opt_include
+      end
+
+      args << "sasl" if build.with? "sasl"
+
+      system "./build.sh", *args
+    end
+
+    mkdir "src/mongo-tools"
+    cp Dir["src/github.com/mongodb/mongo-tools/bin/*"], "src/mongo-tools/"
+
+    args = %W[
+      --prefix=#{prefix}
+      -j#{ENV.make_jobs}
+      --osx-version-min=#{MacOS.version}
+    ]
+
+    args << "CC=#{ENV.cc}"
+    args << "CXX=#{ENV.cxx}"
+
+    args << "--use-sasl-client" if build.with? "sasl"
+    args << "--use-system-boost" if build.with? "boost"
+    args << "--use-new-tools"
+    args << "--disable-warnings-as-errors" if MacOS.version >= :yosemite
+
+    if build.with? "openssl"
+      args << "--ssl"
+
+      args << "CCFLAGS=-I#{Formula["openssl"].opt_include}"
+      args << "LINKFLAGS=-L#{Formula["openssl"].opt_lib}"
+    end
+
+    scons "install", *args
+
+    (buildpath+"mongod.conf").write mongodb_conf
+    etc.install "mongod.conf"
```

Comment:
This line and the above can be shorted to `(etc/"mongod.conf").write mongodb_conf`